### PR TITLE
Remove departed ingesters from routing table

### PR DESCRIPTION
**Pull request by nadav-govari**
### Description

Ingesters that leave the cluster aren't removed from the routing table. We were relying on the ingester pool to not select them. If these ingesters later rejoin the cluster, the old entry is still there- representing stale data. It's only cleared when the control plane inits shards on those ingesters, or they broadcast their first capacity update (both can take seconds). This clears the entry on departure to prevent that edge case.

### How was this PR tested?

Unit tests.

### CODE REVIEW

File: quickwit/quickwit-indexing/src/actors/indexing_service.rs [hunk 1]
```diff
@@ -777,6 +777,7 @@ impl IndexingService {
         Ok(Some(merge_planner_mailbox))
     }
 
+    //noinspection ALL
```
Comment (guilload) [inline quickwit/quickwit-indexing/src/actors/indexing_service.rs]: :)

File: quickwit/quickwit-ingest/src/ingest_v2/mod.rs [hunk 2]
```diff
@@ -96,6 +97,14 @@ impl IngesterPoolEntry {
 
 pub type IngesterPool = Pool<NodeId, IngesterPoolEntry>;
 
+/// Published when an ingester leaves the cluster.
+#[derive(Debug, Clone)]
+pub struct IngesterDeparture {
```
Comment (guilload) [inline quickwit/quickwit-ingest/src/ingest_v2/mod.rs]:
```suggestion
pub struct IngesterLeft {
```

**Review by guilload — APPROVED** [review]

**Comment by nadav-govari** [comment]
On hold as we discuss a more thorough, more complicated fix

**Comment by nadav-govari** [comment]
Superseded by https://github.com/quickwit-oss/quickwit/pull/6680
